### ISSUE TITLE
Separate Strong/Weak Laws for MonadFilter/MonadCombine

### ISSUE THREAD

**Issue by mikejcurry** [typelevel/cats#694]
#693 and #687 refer to Strong and Weak Laws for `MonadFilter`/`MonadCombine`. In #687 the `MonadCombine` instance for `XorT` was commented out as it was unused. It was unused due to the fact that the `MonadCombine` instance for `XorT` could not meet some of the laws for `MonadFilter`, namely: right absorbtion, right distributivity, and left distributivity. #693 will make a similar change for `OptionT` and reduce the available instance from `MonadCombine` to `Monad`.

It is suggested that these need to be separated out into Weak and Strong Laws, at which point an instance, supporting the appropriate weaker laws, can be reinstated for both `XorT` and `OptionT`.

**Comment by kailuowang** [comment]
Some of the problematic laws are in `AlternativeLaws`, which is part of the #1210 refactor. Hence, blocked by #1210

**Comment by kailuowang** [comment]
No longer needed due to #1616, moved out of the 1.0.0-MF, will close once mtl removal plan is completed.
